sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubOptions.cs

```diff
@@ -132,6 +132,27 @@ public int MaxEventBatchSize
             }
         }
 
+        private int _targetUnprocessedEventThreshold;
+
+        /// <summary>
+        /// Get or sets the target unprocessed events per worker for eventhub functions. This is used in target based scaling to override the default behavior of using maxEventBatchSize as the target value.
+        /// If TargetUnprocessedEventThreshold is set, the total unprocessed event count will be divided by this value to determine the number of instances,
+        /// and then we will round that up to an instance count that creates a balanced partition distribution.
+        /// </summary>
+        public int TargetUnprocessedEventThreshold
```

Comment:
@JoshLove-msft :  This would also require updates in the [EventHubExtensionConfigProvider](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubExtensionConfigProvider.cs), right?     

@chiangvincent:  Please also include this in bindings tests where we exercise the other config.

Reply:
I don't believe so. The binding should work automatically based on this [line](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Config/EventHubWebJobsBuilderExtensions.cs#L83). We have special logic for some properties that we wanted to have backwards compat support for in EventHubWebJobsBuilderExtensions. Also properties that are not able to be bound automatically by the extensions framework, e.g. the WebProxy require special handling, but this one shouldn't need anything special. We should definitely add tests to confirm though.

Reply:
Having said that, where is this property actually used? I don't see any code consuming it.

Reply:
> Having said that, where is this property actually used? I don't see any code consuming it.

It'll be picked up by the service-side scale controller and not used within the bindings directly.   That said, I have no idea how that actually happens - @chiangvincent?

Reply:
Created a teams chat with scale controller PR -- will also create a corresponding PR in the sdk-for-net repo

Reply:
Added unit tests for configuration checks

Reply:
Tests are failing due to the validation that ensures the value is greater than 0. Is there a default value that should apply here other than 0? Otherwise, should this property be nullable?

Reply:
+1 for nullable over using `0` as a magic number to represent unset.

Reply:
Done, made `TargetUnprocessedEventThreshold` nullable in the options class.